src/software/network_log_listener_main.cpp

```diff
@@ -0,0 +1,69 @@
+#include <boost/program_options.hpp>
+#include <iostream>
+#include <numeric>
+
+#include "proto/robot_log_msg.pb.h"
+#include "proto/tbots_software_msgs.pb.h"
+#include "shared/constants.h"
+#include "shared/parameter/cpp_dynamic_parameters.h"
+#include "software/networking/threaded_proto_udp_listener.h"
+
+/*
+ * This standalone program listens for RobotLog protos on the specified ip address
+ * and logs them
+ */
+
+void logFromNetworking(TbotsProto::RobotLog log)
+{
+    LEVELS level(INFO);
+
+    if (TbotsProto::LogLevel_Name(log.log_level()) == "DEBUG")
+    {
+        level = DEBUG;
+    }
+    else if (TbotsProto::LogLevel_Name(log.log_level()) == "WARNING" ||
+             TbotsProto::LogLevel_Name(log.log_level()) == "FATAL")
+    {
+        // log FATAL as WARNING to prevent program from exiting
+        // note that the level of the RobotLog will itself be printed
+        level = WARNING;
+    }
+
+    LOG(level) << "[ROBOT " << log.robot_id() << " " << LogLevel_Name(log.log_level())
+               << "]"
+               << "[" << log.file_name() << ":" << log.line_number()
+               << "]: " << log.log_msg() << std::endl;
+}
+
+int main(int argc, char **argv)
+{
+    // load command line arguments
+    auto args           = std::make_shared<NetworkLogListenerMainCommandLineArgs>();
+    bool help_requested = args->loadFromCommandLineArguments(argc, argv);
+
+    auto logWorker               = g3::LogWorker::createLogWorker();
+    auto colour_cout_sink_handle = logWorker->addSink(
+        std::make_unique<ColouredCoutSink>(false), &ColouredCoutSink::displayColouredLog);
```

Comment:
Is print_detailed false because the log that it's logging already has a newline? Can we have a bit more documentation on print_detailed?